include/pmacc/math/vector/Vector.hpp

```diff
@@ -201,138 +152,115 @@ namespace pmacc
              * @param value Value which is set for all dimensions
              * @return new Vector<...>
              */
-            HDINLINE
-            static constexpr Vector create(ParamType value)
+            static constexpr Vector create(T_Type const& value)
             {
-                Vector result;
-                for(uint32_t i = 0u; i < dim; i++)
-                    result[i] = value;
+                Vector result([=](uint32_t const) { return value; });
 
                 return result;
             }
 
-            HDINLINE const Vector& toRT() const
-            {
-                return *this;
-            }
-
-            HDINLINE Vector& toRT()
+            constexpr Vector toRT() const
             {
                 return *this;
             }
 
-            HDINLINE Vector revert()
+            constexpr Vector revert() const
             {
-                Vector invertedVector;
+                Vector invertedVector{};
                 for(uint32_t i = 0u; i < dim; i++)
                     invertedVector[dim - 1 - i] = (*this)[i];
 
                 return invertedVector;
             }
 
-            HDINLINE Vector& operator=(const Vector&) = default;
+            constexpr Vector& operator=(const Vector&) = default;
 
-            template<typename T_OtherNavigator, typename T_OtherStorage>
-            HDINLINE Vector& operator=(const Vector<type, dim, T_OtherNavigator, T_OtherStorage>& rhs)
+            constexpr Vector operator-() const
             {
+                auto result = Vector<T_Type, T_dim>{};
                 for(uint32_t i = 0u; i < dim; i++)
-                    (*this)[i] = rhs[i];
-                return *this;
-            }
-
-            HDINLINE
-            type& operator[](const uint32_t idx)
-            {
-                return Storage::operator[](Navigator::operator()(idx));
-            }
-
-            HDINLINE
-            const type& operator[](const uint32_t idx) const
-            {
-                return Storage::operator[](Navigator::operator()(idx));
-            }
-
-            HDINLINE type& x()
-            {
-                return (*this)[0];
-            }
-
-            HDINLINE type& y()
-            {
-                PMACC_CASSERT_MSG(math_Vector__access_to_y_is_not_allowed_for_DIM_lesser_than_2, dim >= 2u);
-                return (*this)[1];
+                    result[i] = -(*this)[i];
+                return result;
```

Comment:
Can be written as `return Vector([=](uint32_t const i) constexpr { return -(*this)[i]; })`